[recast] Imported target includes non-existent path

**Describe the bug**
`recast` defines a target that adds non-existent paths to its `INTERFACE_INCLUDE_DIRECTORIES` property.

**Environment**
- OS: Windows
- Compiler: MSVC v143

**To Reproduce**
Steps to reproduce the behavior:
1. Clone https://github.com/phisko/vcpkg_recast_fail
2. `mkdir build`
3. `cd build`
4. Run cmake with `-DCMAKE_TOOLCHAIN_FILE` pointing to `vcpkg.cmake`
5. See error

**Expected behavior**
CMake runs without failure

**Failure logs**
```
$ cmake .. -DCMAKE_TOOLCHAIN_FILE=/c/git/vcpkg/scripts/buildsystems/vcpkg.cmake
-- Building for: Visual Studio 17 2022
CMake Warning (dev) at CMakeLists.txt:1 (project):
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Running vcpkg install
Detecting compiler hash for triplet x64-windows...
The following packages will be built and installed:
    recast[core]:x64-windows -> 1.5.1#5
  * vcpkg-cmake[core]:x64-windows -> 2022-12-22
  * vcpkg-cmake-config[core]:x64-windows -> 2022-02-06#1
Additional packages (*) will be modified to complete this operation.
Restored 3 package(s) from C:\Users\nicol\AppData\Local\vcpkg\archives in 91.26 ms. Use --debug to see more details.
Installing 1/3 vcpkg-cmake-config:x64-windows...
Elapsed time to handle vcpkg-cmake-config:x64-windows: 9.086 ms
Installing 2/3 vcpkg-cmake:x64-windows...
Elapsed time to handle vcpkg-cmake:x64-windows: 9.755 ms
Installing 3/3 recast:x64-windows...
Elapsed time to handle recast:x64-windows: 34.39 ms
Total install time: 144.6 ms
recast provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(unofficial-recast CONFIG REQUIRED)
    target_link_libraries(main PRIVATE recast)

-- Running vcpkg install - done
-- The C compiler identification is MSVC 19.34.31942.0
-- The CXX compiler identification is MSVC 19.34.31942.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning (dev) in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as

    cmake_minimum_required(VERSION 3.26)

  should be added at the top of the file.  The version specified may be lower
  if you wish to support older CMake versions for this project.  For more
  information run "cmake --help-policy CMP0000".
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring done (7.5s)
CMake Error in CMakeLists.txt:
  Imported target "recast" includes non-existent path

    "C:/git/tmp/build/vcpkg_installed/x64-windows/include/recast/Detour/Include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in CMakeLists.txt:
  Imported target "recast" includes non-existent path

    "C:/git/tmp/build/vcpkg_installed/x64-windows/include/recast/Detour/Include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in CMakeLists.txt:
  Imported target "recast" includes non-existent path

    "C:/git/tmp/build/vcpkg_installed/x64-windows/include/recast/Detour/Include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in CMakeLists.txt:
  Imported target "recast" includes non-existent path

    "C:/git/tmp/build/vcpkg_installed/x64-windows/include/recast/Detour/Include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in CMakeLists.txt:
  Imported target "recast" includes non-existent path

    "C:/git/tmp/build/vcpkg_installed/x64-windows/include/recast/Detour/Include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in CMakeLists.txt:
  Imported target "recast" includes non-existent path

    "C:/git/tmp/build/vcpkg_installed/x64-windows/include/recast/Detour/Include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in CMakeLists.txt:
  Imported target "recast" includes non-existent path

    "C:/git/tmp/build/vcpkg_installed/x64-windows/include/recast/Detour/Include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in CMakeLists.txt:
  Imported target "recast" includes non-existent path

    "C:/git/tmp/build/vcpkg_installed/x64-windows/include/recast/Detour/Include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in CMakeLists.txt:
  Imported target "recast" includes non-existent path

    "C:/git/tmp/build/vcpkg_installed/x64-windows/include/recast/Detour/Include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in CMakeLists.txt:
  Imported target "recast" includes non-existent path

    "C:/git/tmp/build/vcpkg_installed/x64-windows/include/recast/Detour/Include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in CMakeLists.txt:
  Imported target "recast" includes non-existent path

    "C:/git/tmp/build/vcpkg_installed/x64-windows/include/recast/Detour/Include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in CMakeLists.txt:
  Imported target "recast" includes non-existent path

    "C:/git/tmp/build/vcpkg_installed/x64-windows/include/recast/Detour/Include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



-- Generating done (0.0s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

**Additional context**
This is my first time trying to work with vcpkg so unfortunately I haven't been able to find a fix for this. Looking at [the recast port CMakeLists](https://github.com/microsoft/vcpkg/blob/master/ports/recast/CMakeLists.txt#L45) it seems clear that there's an incoherence between these lines:
```
  $<INSTALL_INTERFACE:include/Detour/Include>
  $<INSTALL_INTERFACE:include/DetourCrowd/Include>
  $<INSTALL_INTERFACE:include/DetourTileCache/Include>
  $<INSTALL_INTERFACE:include/Recast/Include>
  $<INSTALL_INTERFACE:include/DebugUtils/Include>
```

and these:
```
  install(DIRECTORY DebugUtils/Include/ DESTINATION include/recast/DebugUtils)
  install(DIRECTORY Detour/Include/ DESTINATION include/recast/Detour)
  install(DIRECTORY DetourCrowd/Include/ DESTINATION include/recast/DetourCrowd)
  install(DIRECTORY DetourTileCache/Include/ DESTINATION include/recast/DetourTileCache)
  install(DIRECTORY Recast/Include/ DESTINATION include/recast/Recast)
```

I was able to fix the issue with [this commit](https://github.com/phisko/vcpkg/commit/8ca5cf8b1762762ba25e4072ef9daf16c9c18008). PR incoming

@phisko, thanks for posting this issue, I followed your steps to reproduce the behavior with no success but I got a different error than yours, I wanted to check with you if I went wrong?

1. git clone https://github.com/phisko/vcpkg_recast_fail
````
PS F:\Feature-test> git clone https://github.com/phisko/vcpkg_recast_fail
Cloning into 'vcpkg_recast_fail'...
remote: Enumerating objects: 8, done.
remote: Counting objects: 100% (8/8), done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 8 (delta 0), reused 8 (delta 0), pack-reused 0
Receiving objects: 100% (8/8), done.
````
2. mkdir build
````
PS F:\Feature-test> mkdir build


    Directory: F:\Feature-test


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----         2/27/2023   3:20 PM                build
````
3. cd build
````
PS F:\Feature-test> cd build
PS F:\Feature-test\build> pwd

Path
----
F:\Feature-test\build
````
4. Run cmake with -DCMAKE_TOOLCHAIN_FILE pointing to vcpkg.cmake
````
PS F:\Feature-test\vcpkg_recast_fail\build> F:\Feature-test\vcpkg_recast_fail\vcpkg\downloads\tools\cmake-3.25.1-windows\cmake-3.25.1-windows-i386\bin\cmake .. -G "Visual Studio 16 2019" -A x64  -DVCPKG_TARGET_TRIPLET=x64-windows -DVCPKG_BUILD_TYPE=debug  -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/vcpkg_recast_fail/vcpkg/scripts/buildsystems/vcpkg.cmake
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19045.
-- The C compiler identification is MSVC 19.29.30147.0
-- The CXX compiler identification is MSVC 19.29.30147.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:846 (_find_package):
  Could not find a package configuration file provided by "unofficial-recast"
  with any of the following names:

    unofficial-recastConfig.cmake
    unofficial-recast-config.cmake

  Add the installation prefix of "unofficial-recast" to CMAKE_PREFIX_PATH or
  set "unofficial-recast_DIR" to a directory containing one of the above
  files.  If "unofficial-recast" provides a separate development package or
  SDK, be sure it has been installed.
Call Stack (most recent call first):
  CMakeLists.txt:5 (find_package)


CMake Warning (dev) in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as

    cmake_minimum_required(VERSION 3.25)

  should be added at the top of the file.  The version specified may be lower
  if you wish to support older CMake versions for this project.  For more
  information run "cmake --help-policy CMP0000".
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring incomplete, errors occurred!
See also "F:/Feature-test/vcpkg_recast_fail/build/CMakeFiles/CMakeOutput.log".
````

--- a/ports/recast/CMakeLists.txt
+++ b/ports/recast/CMakeLists.txt
@@ -41,12 +41,12 @@ target_include_directories(
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/DetourCrowd/Include>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/DetourTileCache/Include>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Recast/Include>
-  $<INSTALL_INTERFACE:include>
-  $<INSTALL_INTERFACE:include/Detour/Include>
-  $<INSTALL_INTERFACE:include/DetourCrowd/Include>
-  $<INSTALL_INTERFACE:include/DetourTileCache/Include>
-  $<INSTALL_INTERFACE:include/Recast/Include>
-  $<INSTALL_INTERFACE:include/DebugUtils/Include>
+  $<INSTALL_INTERFACE:include/recast/>
+  $<INSTALL_INTERFACE:include/recast/Detour/Include>
+  $<INSTALL_INTERFACE:include/recast/DetourCrowd/Include>
+  $<INSTALL_INTERFACE:include/recast/DetourTileCache/Include>
+  $<INSTALL_INTERFACE:include/recast/Recast/Include>
+  $<INSTALL_INTERFACE:include/recast/DebugUtils/Include>
 )
 
 install(
@@ -59,9 +59,9 @@ install(
 install(EXPORT unofficial-recast-config DESTINATION share/unofficial-recast)
 
 if(NOT DISABLE_INSTALL_HEADERS)
-  install(DIRECTORY DebugUtils/Include/ DESTINATION include/recast/DebugUtils)
-  install(DIRECTORY Detour/Include/ DESTINATION include/recast/Detour)
-  install(DIRECTORY DetourCrowd/Include/ DESTINATION include/recast/DetourCrowd)
-  install(DIRECTORY DetourTileCache/Include/ DESTINATION include/recast/DetourTileCache)
-  install(DIRECTORY Recast/Include/ DESTINATION include/recast/Recast)
+  install(DIRECTORY DebugUtils/Include/ DESTINATION include/recast/DebugUtils/Include)
+  install(DIRECTORY Detour/Include/ DESTINATION include/recast/Detour/Include)
+  install(DIRECTORY DetourCrowd/Include/ DESTINATION include/recast/DetourCrowd/Include)
+  install(DIRECTORY DetourTileCache/Include/ DESTINATION include/recast/DetourTileCache/Include)
+  install(DIRECTORY Recast/Include/ DESTINATION include/recast/Recast/Include)
 endif()

--- a/ports/recast/vcpkg.json
+++ b/ports/recast/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "recast",
   "version": "1.5.1",
-  "port-version": 5,
+  "port-version": 6,
   "description": "Navigation-mesh Toolset for Games",
   "homepage": "https://github.com/recastnavigation/recastnavigation",
   "license": "Zlib",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6798,7 +6798,7 @@
     },
     "recast": {
       "baseline": "1.5.1",
-      "port-version": 5
+      "port-version": 6
     },
     "recycle": {
       "baseline": "6.0.0",

--- a/versions/r-/recast.json
+++ b/versions/r-/recast.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "f6f99b923ad8fbbee1aafd0deb361f8cf7eb7bd6",
+      "version": "1.5.1",
+      "port-version": 6
+    },
+    {
       "git-tree": "fe2b42ef737e0b51a58bb9f56340e324ab8f82c3",
       "version": "1.5.1",
       "port-version": 5

